src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Workloads/WpfWorkloadDescriptorDataSource.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Concurrent;
+using System.Threading.Tasks.Dataflow;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Workloads
+{
+    [Export(typeof(IWpfWorkloadDescriptorDataSource))]
+    [AppliesTo(ProjectCapability.DotNet)]
+    internal class WpfWorkloadDescriptorDataSource : ChainedProjectValueDataSourceBase<ISet<WorkloadDescriptor>>, IWpfWorkloadDescriptorDataSource
+    {
+        private static readonly string s_webWorkloadName = "Web";
+        private static readonly string s_webComponentId = "Microsoft.VisualStudio.Component.Web";
+        private static readonly WorkloadDescriptor s_webWorkload = new(s_webWorkloadName, s_webComponentId);
+
+        private readonly ConcurrentHashSet<string> _componentIdsDetected = new();
```

Comment:
Would `_componentsIdsRequired` be a better name? If looks like we're checking if the web component is _needed_, not if it is _already_ present.

Reply:
changed